hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/ReadableNodeStoreImpl.java

```diff
@@ -79,9 +79,9 @@ protected <T extends ReadableKVState<EntityNumber, Node>> T nodesState() {
 
     @NonNull
     public List<EntityNumber> keys() {
-        final var size = sizeOfState();
         final var keys = new ArrayList<EntityNumber>();
-        for (int i = 0; i < size; i++) {
+        final long highestExclusive = entityIdStore.peekAtNextNodeId();
```

Comment:
nit: Is there a reason to not have a method, `getHighestNodeId`, instead of looking at next node id?